### ISSUE TITLE
ignore-methods does not prevent removal of awaited method calls

### ISSUE THREAD

**Issue by nwalker-et** [stryker-mutator/stryker-net#3006]
**Describe the bug**
If one adds a method to the `ignore-methods` list, Stryker will ignore mutations that modify or remove calls to the method. However, if the method is awaited, then Stryker will *not* ignore mutations that remove the await statement invoking the method.

**Logs**
[log-20240808.txt](https://github.com/user-attachments/files/16550066/log-20240808.txt)

<img width="430" alt="image" src="https://github.com/user-attachments/assets/7341d0a9-83e3-42c9-982d-5af6412f9d7d">

**Expected behavior**
I expect directly awaited asynchronous method calls to be treated the same as synchronous method calls. Namely, the await statement containing the invocation shouldn't be removed if the method has been ignored.

In my opinion, this is distinct from when a task is stored in a variable and not immediately awaited. In those cases, I think it is fair to not expect Stryker to know that a given await statement is tied to a previous method call.

**Desktop (please complete the following information):**
 - OS: Windows
 - Type of project: Microsoft.NET.Sdk
 - Framework Version: net8.0
 - Stryker Version: 4.1.0

**Additional context**
Here is the source code I've used to reproduce this issue:
<details>

`AwaitIssueDemo.cs`
```csharp
namespace LibraryProject;

public class AwaitIssueDemo {
  public string CallMethod() {
    Thread.Sleep(10);
    Task.Delay(10);
    return "value";
  }

  public async Task<string> CallMethodAsync() {
    await Task.Delay(10);
    return "value";
  }
}
```

`AwaitIssueDemoTests.cs`
```csharp
namespace LibraryProject.Test;

public class AwaitIssueDemoTests {
  private AwaitIssueDemo exampleClass = new AwaitIssueDemo();

  [Fact]
  public void CallMethod_ReturnsString() {
    Assert.Equal("value", exampleClass.CallMethod());
  }

  [Fact]
  public async Task IgnoreTaskAsync_ReturnsString() {
    Assert.Equal("value", await exampleClass.CallMethodAsync());
  }
}
```

`stryker-config.json`
```json
{
  "stryker-config": {
    "ignore-methods": [
      "Task.Delay",
      "Thread.Sleep"
    ]
  }
}
```

</details>

**Comment by dupdob** [comment]
Thanks for reporting this issue. I agree with you that async expressions should be handled for ignore methods logic.
I will open a PR with a fix soon.

**Comment by nwalker-et** [comment]
> Thanks for reporting this issue. I agree with you that async expressions should be handled for ignore methods logic. I will open a PR with a fix soon.

I have a coworker who is also most of the way towards making a PR for this, so unless you've already done the work, I would be fine waiting for him to make the PR. Either way is great though, thanks for your quick response!

**Comment by dupdob** [comment]
Sorry
PR is already in draft.

**Comment by nwalker-et** [comment]
No worries!

**Comment by nwalker-et** [comment]
Should this issue be closed since #3010 got merged?
